test/e2e/report_dynamic_input_data_test.go

```diff
@@ -163,3 +169,129 @@ func testReportsProduceData(t *testing.T, testReportingFramework *reportingframe
 		})
 	}
 }
+
+func testReportIsDeletedWhenNoDeps(t *testing.T, testReportingFramework *reportingframework.ReportingFramework) {
+	// cron schedule to run every minute
+	cronSchedule := &metering.ReportSchedule{
+		Period: metering.ReportPeriodCron,
+		Cron: &metering.ReportScheduleCron{
+			Expression: "*/1 * * * *",
+		},
+	}
+	var foundOnce bool
+	expiration := &metav1.Duration{Duration: 30.0 * time.Second}
+	// If a report is in dependency unmet state it won't be deleted and it can take awhile for import to happen.
+	// 4 minutes seems to be enough time for that setup, and the delete, to reliably happen.
+	waitTimeWhileCheckDeletion := 4 * time.Minute
+
+	report := testReportingFramework.NewSimpleReport(
+		"report-should-delete",
+		"namespace-memory-request",
+		cronSchedule,
+		nil,
+		nil,
+		expiration,
+	)
+	t.Logf("creating report %s and waiting %s to finish", report.Name, waitTimeWhileCheckDeletion)
+	createErr := testReportingFramework.CreateMeteringReport(report)
+	if createErr != nil {
+		assert.Failf(t, "Creating report should not have an error", createErr.Error())
+	}
+	err := wait.Poll(time.Second*5, waitTimeWhileCheckDeletion, func() (bool, error) {
+		_, err := testReportingFramework.GetMeteringReport("report-should-delete")
+		// `foundOnce` provides a latch that allows us to wait for report creation then exit on deletion
+		if err == nil {
```

Comment:
Same here, we should use `if apierrors.IsAlreadyExists(err) { foundOnce = true }`.